src/main/kotlin/com/mparticle/kits/UrbanAirshipKit.kt

```diff
@@ -61,7 +62,9 @@ class UrbanAirshipKit : KitIntegration(), KitIntegration.PushListener, KitIntegr
     }
 
     override fun setOptOut(optedOut: Boolean): List<ReportingMessage> {
-        UAirship.shared().isDataCollectionEnabled = !optedOut
+        UAirship.shared().privacyManager.setEnabledFeatures(
+            if (optedOut) PrivacyManager.Feature.ALL else PrivacyManager.Feature.NONE
```

Comment:
Is this logic backwards? If I'm reading this correctly, we enable all features if they opt out and no features if they opt in?

Reply:
Yes, so before, it was working internally like this:

 public void setDataCollectionEnabled(boolean enabled) {
        if (enabled) {
            this.privacyManager.setEnabledFeatures(PrivacyManager.FEATURE_ALL);
        } else {
            this.privacyManager.setEnabledFeatures(PrivacyManager.FEATURE_NONE);
        }
    }

Reply:
Thanks for the explanation!

Reply:
I don't fully follow @Mansi-mParticle 's comment.  Where was this code "before" and where "internally"?

Reply:
This code was from Airship version 17.7.1 and is related to the internal Airship class.

Reply:
This feels backwards to me as well, if you are opted out you should disable all features?  The above example you use enable but this is the inverse of it so I think it should be switched?

Reply:
Before:
`UAirship.shared().isDataCollectionEnabled = !optedOut`
- false: data collection enabled
- true: data collection disabled

After:
`if (optedOut) PrivacyManager.Feature.ALL else PrivacyManager.Feature.NONE`
- false: data collection disabled (none enabled))
- true: data collection enabled (all enabled)

Reply:
@rlepinski  Can you please confirm these changes?